### PR TITLE
feat: display merkl apr in tooltip

### PR DESCRIPTION
**Pull request by Chef-Yogi**
---

## PR-Codex overview
This PR updates various components in the codebase to include `merklApr` and `lpAddress` properties for displaying Merkl APR information.

### Detailed summary
- Added `merklApr` property to components for displaying Merkl APR information
- Updated components to include `lpAddress` property
- Added `useMerklInfo` hook for fetching Merkl APR data

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

File: packages/widgets-internal/farm/components/MerklNotice/index.tsx [hunk 1]
```diff
@@ -12,15 +14,39 @@ type MerklNoticeContentProps = {
...
 export const MerklNoticeContent: React.FC<MerklNoticeContentProps> = ({
   linkColor = "primary",
   merklLink,
   hasFarm,
+  chainId,
+  lpAddress,
 }) => {
   const { t } = useTranslation();
 
+  const { data } = useQuery({
```
Comment (chefjackson) [inline packages/widgets-internal/farm/components/MerklNotice/index.tsx:30]: For widgets we still want to keep it UI related only. Can we do the network IO in the web application and feed in the data to the component?

Comment (Chef-Yogi) [reply packages/widgets-internal/farm/components/MerklNotice/index.tsx:30]: sure, no problem

Comment (Chef-Yogi) [reply packages/widgets-internal/farm/components/MerklNotice/index.tsx]: Done

File: apps/web/src/hooks/useMerkl.tsx [hunk 2]
```diff
@@ -203,3 +203,27 @@ export default function useMerkl(poolAddress: string | null) {
     [claimTokenReward, hasMerkl, isTxPending, rewardsPerToken],
   )
 }
+
+export const useMerklApr = (lpAddress?: Address, chainId?: number, isMerklPool?: boolean) => {
```
Comment (chefjackson) [inline apps/web/src/hooks/useMerkl.tsx:207]: how about we combine this hook with `useMerklInfo`?

Comment (Chef-Yogi) [reply apps/web/src/hooks/useMerkl.tsx:207]: sure

File: apps/web/src/hooks/useMerkl.tsx [hunk 3]
```diff
@@ -29,9 +29,30 @@ export function useMerklInfo(poolAddress: string | null): {
...
+    queryKey: ['merklAprData', chainId, poolAddress],
+    queryFn: async () => {
+      try {
+        const resp = await (
+          await fetch(`https://api.angle.money/v2/merkl?chainIds[]=${chainId}&AMMs[]=pancakeswapv3`)
+        ).json()
+        return resp
+      } catch (error) {
+        console.error('Fetch merklAprData Error: ', error)
+        return true
```
Comment (ChefJerry) [inline apps/web/src/hooks/useMerkl.tsx:46]: why return `true`

Comment (Chef-Yogi) [reply apps/web/src/hooks/useMerkl.tsx:46]: Oh, let me remove this

Comment (Chef-Yogi) [reply apps/web/src/hooks/useMerkl.tsx:46]: nice catch , and thanks

Comment (Chef-Yogi) [reply apps/web/src/hooks/useMerkl.tsx]: should return null here

File: apps/web/src/hooks/useMerkl.tsx [hunk 4]
```diff
@@ -29,9 +29,30 @@ export function useMerklInfo(poolAddress: string | null): {
...
   const { account, chainId } = useAccountActiveChain()
 
+  const { data: merklData } = useQuery({
+    queryKey: ['merklAprData', chainId, poolAddress],
+    queryFn: async () => {
+      try {
+        const resp = await (
+          await fetch(`https://api.angle.money/v2/merkl?chainIds[]=${chainId}&AMMs[]=pancakeswapv3`)
+        ).json()
+        return resp
```
Comment (ChefJerry) [inline apps/web/src/hooks/useMerkl.tsx:43]: the best practice of `fetch` is, always check `response.ok` is `true`

```ts

const response = await fetch()

if (response.ok) return response.json()

throw response

```

Comment (Chef-Yogi) [reply apps/web/src/hooks/useMerkl.tsx:43]: Ok, let me adjust it

**Review by chefjackson — APPROVED** [review]

**Pull request merged**